nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-services/src/main/java/org/apache/nifi/services/iceberg/HiveCatalogService.java

```diff
@@ -69,28 +65,43 @@ protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
         return PROPERTIES;
     }
 
-    private HiveCatalog catalog;
-
     @Override
     protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
 
         final List<ValidationResult> problems = new ArrayList<>();
-        String configMetastoreUri = null;
-        String configWarehouseLocation = null;
+        boolean configMetastoreUriPresent = false;
+        boolean configWarehouseLocationPresent = false;
```

Comment:
Not related to the actual change just noticed now that the expression language support related config of the properties can be enchanced.

In my view both METASTORE_URI and WAREHOUSE_LOCATION can use

    .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
    
  since flow files are not used when they are evaluated.
  (The same comment applies for WAREHOUSE_PATH in HadoopCatalogService.)